File: lsclient.client/src/app/models/jobRequestPayload.ts
import { TrucksPayload } from "./TrucksPayload";

export interface JobRequestPayload {
  jobRequestID: string;            
  pickupLocation: string;         
  deliveryLocation: string;       
  cargoDescription: string;      
  containerNumber: string; 
  status: string;                 
  truckID: string;                
  customerID: string;
  contractId:string;

}

export interface RequestWithPrice {
invoiceNumber?: number;
  jobRequestID: string;            
  pickupLocation: string;         
  deliveryLocation: string;       
  cargoDescription: string;      
  containerNumber: string; 
  status: string;                 
  priceAgreementID: string;
  truckType: string;                        
  truckID: string;  
  driverID: string;                              
  requestType: string;                
  customerID: string;
  contractId:string;
  firstDepositAmount?: number;
  companyAdvanceAmountRequred?:  number;
  // Properties that may need default values for binding
  requestedPrice?: number;  // Use `?` to make it optional initially
  acceptedPrice?: number;
  customerPrice?: number;
  companyID : string;

}

File: lsclient.Server/Models/DataPayloads/JobRequestPayload.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace lsclient.Server.Models.DataPayloads
{
    public class JobRequestPayload
    {
         
            public string? JobRequestID { get; set; }
            public string? PickupLocation { get; set; }
            public string? DeliveryLocation { get; set; }
            public string? CargoDescription { get; set; }
            public string? ContainerNumber { get; set; } 

            public string? Status { get; set; }  
            public string? PriceAgreementID { get; set; }
         
            public string? TruckID { get; set; }
            public string? CustomerID { get; set; }


         
    }
}
